scipy/sparse/linalg/_dsolve/linsolve.py

```diff
@@ -51,6 +51,17 @@ def use_solver(**kwargs):
     sure that the matrix fulfills this, pass ``assumeSortedIndices=True``
     to gain some speed.
 
+    Example
+    ------
+    >>> from scipy.sparse.linalg import use_solver, spsolve
+    >>> from scipy.sparse import csc_matrix
+    >>> R = np.random.randn(5, 5)
+    >>> A = csc_matrix(R)
+    >>> b = np.random.randn(5)
+    >>> use_solver(useUmfpack=True, assumeSortedIndices=True) # use UMFPACK over SuperLU.
```

Comment:
This fails in CI (the job with `refguide` in it) with: `NameError: name 'umfpack' is not defined`. Scikit-umfpack is an optional dependency, which is not installed there. Given that umfpack is the default if installed, I think it makes more sense to use `useUmfpack=True` in the example. And that won't run into a problem with the dependency not being available.

Note that `use_solver` is a bit weird as it changes global state, so at the end of the example please reset to default to `True` to avoid affecting other tests/docs.